iac-extensions/spring-config/src/test/java/org/sonar/iac/springconfig/parser/SpringConfigParserTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.springconfig.parser;
+
+import org.junit.jupiter.api.Test;
+import org.sonar.iac.common.extension.ParseException;
+import org.sonar.iac.common.extension.visitors.InputFileContext;
+import org.sonar.iac.common.yaml.YamlLanguage;
+import org.sonar.iac.springconfig.tree.api.File;
+import org.sonar.iac.springconfig.tree.api.SpringConfig;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.sonar.iac.common.testing.IacTestUtils.createInputFileContextMock;
+
+class SpringConfigParserTest {
+  private final SpringConfigParser parser = new SpringConfigParser();
+
+  private final InputFileContext inputFilePropertiesContext = createInputFileContextMock("application.properties");
+  private final InputFileContext inputFileYamlContext = createInputFileContextMock("application.yaml", YamlLanguage.KEY);
+
+  @Test
+  void shouldParseSimpleYaml() {
+    String source = """
+      a:
+        b: c""";
+    File file = (File) parser.parse(source, inputFileYamlContext);
+    assertThat(file.profiles().get(0).properties()).hasSize(1);
+  }
+
+  @Test
+  void shouldParseSimpleProperties() {
+    File tree = (File) parser.parse("foo=bar", inputFilePropertiesContext);
+    assertThat(tree.profiles().get(0).properties()).hasSize(1);
+  }
+
+  @Test
+  void shouldThrowParseExceptionOnEmptyYamlFile() {
+    assertThatThrownBy(() -> parser.parse("", inputFileYamlContext))
```

Comment:
I'm slightly surprised that the empty file throws `ParseException` - this will raise issue S2260.
For Kubernetes, Cloudformation and ARM (another YAML parser) there is a pre-validation in `customFilePredicate()` that the file contains some identifier (= is not empty).
For Spring config we don't have such pre-validation. 
We should not fail on an empty file. I can imagine that someone committed an empty file or created a file (in SonarLint context some day in the future) and in that cases we should not raise S2260.

Reply:
Actually, that's implemented in the `IacSensor` itself. If you look at line 226 there's this snippet:
```
      if (EMPTY_FILE_CONTENT_PATTERN.matcher(content).matches()) {
        return;
      }
```
Basically we don't parse empty files at all.

Reply:
OK, I missed that.

Reply:
I will write a small comment to clarify this behavior.